### ISSUE TITLE
`with_defaults` not working (as expected)

### ISSUE THREAD

**Issue by grssnbchr** [jimhester/lintr#288]
Either this does not work as intended or I totally misunderstood it (but then, maybe, the documentation should be more clear). 

When I specify ...

```
lintr::with_defaults(commented_code_linter = NULL,
                     trailing_whitespace_linter = NULL) 
```

... I expect that commented code and lines with trailing whitespace are *not* mentioned by `lintr`. Still, these "issues" are reported. From the documentation: "If the value is NULL the linter is removed."

Also, I find this example in the documentation illogical:

```
# enforce camelCase rather than snake_case
with_defaults(camel_case_linter = NULL,
              snake_case_linter)
```

Shouldn't it be `#enforce snake_case rather than camelCase`? 

I am happy to update the documentation with a pull request, but I suspect something is still not working correctly.

**Comment by grssnbchr** [comment]
Ah, now I get it: The return value of this function call needs to be passed to `lint()`. 

```
lintr::lint("main.Rmd", linters = 
              lintr::with_defaults(
                commented_code_linter = NULL,
                trailing_whitespace_linter = NULL
                )
            )
```

Is this clear from the documentation? I don't think so.
